source/develop/release-management/features/network/sriov-live-migration.md

```diff
@@ -0,0 +1,178 @@
+Related patches can be found on [gerrit topic:sriov_live_migration][3].
+
+## Owner
+
+*   Name: [Ales Musil](https://github.com/almusil)
+
+*   Email: <amusil@redhat.com>
+
+
+## Detailed Description
+
+In order to migrate a VM that is connected to SR-IOV passthrough profile
+we need to first unplug the SR-IOV device from a VM on the source, migrate
+the VM and plug SR-IOV device on the destination. This causes longer downtime
+during which the VM is not connected to the network.
+
+The current implementation allowed to mitigate this downtime
+[Live Migration Support For SRIOV][1], however this approach required manual
+configuration to be done in the guest which is not convenient for large setups. 
+
+In order to achieve minimal downtime the [NET_FAILOVER][4] support was added 
+to Linux kernel. The driver provides automated mechanism which is transparent 
+to the VM, and the user space. It is achieved by `failover` netdev which acts
+as master device and controls two slave devices, the `primary` device being 
+SR-IOV passtrough and the virtio as `standby`. The VM user access network
+through the `failover` device. 
+
+There is still need to provide two interfaces, the SR-IOV passtrough device
+and virtio device, to the affected VM, but the  overall configuration overhead
+has been reduced to selecting suitable vNIC profile pair.
+
+
+## Prerequisites
+
+The implementation requires a [Linux kernel][5], [libvirt][6] and [qemu][7] 
+support with `net_failover`. 
+
+The [Linux kernel][5] support has to be only on guest side. 
+
+## Limitations
+
+* In addition to the SR-IOV NIC, a linux-bridge based (or an OVN physical network in OVS clusters)
+  has to be connected to the source, and the destination host, marked as required.
+
+## Implementation
+
+### Entity Description
+
+vNIC profile will get a new attribute that will point to the failover virtio 
+vNIC profile. This selection will be only available with `Passtrough` and `Migratable`
+selected. 
+
+### User Experience
+
+Upon creation of vNIC profile, the user can enable failover and select 
+the failover network vNIC profile. The `Failover` selection is available 
+only for `Passtroguh` and `Migratable` vNIC profiles.
+
+REST API takes an additional parameter for vNIC called `failover`
+containing id of the failover vNIC profile:
+
+```
+<vnic_profile id="123">
+  <name>vNIC_name</name>
+  <pass_through>
+    <mode>enabled</mode>
+  </pass_through>
+  <migratable>true</migratable>
+  <failover id="456"/>
+</vnic_profile>
+```
+
+VM interface then requires only single vNIC profile that has the `failover` attribute 
+in order to utilize this behavior.
+
+### Engine backend
+
+As engine is responsible for creating XML for a new VM nic, the 
+appropriate XML will be created and provided to Libvirt. 
+
+In order to ensure that the virtio network is available it has to be
+marked as required, and the migration policy needs to be updated.
+
```

Comment:
Do we want to ensure that the guest supports the feature? What should or might happen, if the guest does not support the feature?

Reply:
Good question :)

Reply:
Engine knows kernel version of quest, but IMO I would not put any restriction there. Even if the kernel version has the module it might be disabled/compiled without it.

Reply:
Maybe the backup interface would be invisible or ignored by the guest? If we don't know this yet, we should note this as an open question which we have to check.

Reply:
No guest will see both interface no matter if the failover module is there or not. But only in case of the kernel support it will create the teaming interface automatially.